src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4128,6 +4262,58 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             return (method, results, shouldReturnNotNull);
         }
 
+        private void ApplyMemberPostConditions(
+            ImmutableArray<string> notNullMembers,
+            ImmutableArray<string> notNullWhenTrueMembers,
+            ImmutableArray<string> notNullWhenFalseMembers)
+        {
+            if (notNullWhenTrueMembers.IsEmpty && notNullWhenFalseMembers.IsEmpty)
+            {
+                applyMemberPostConditions(notNullMembers, ref State);
+            }
+            else
+            {
+                Split();
+                applyMemberPostConditions(notNullWhenTrueMembers, ref StateWhenTrue);
+                applyMemberPostConditions(notNullWhenFalseMembers, ref StateWhenFalse);
+            }
+
+            void applyMemberPostConditions(ImmutableArray<string> members, ref LocalState state)
+            {
+                if (members.IsEmpty)
+                {
+                    return;
+                }
+
+                foreach (var memberName in members)
+                {
+                    markMembersAsNotNull(memberName, ref state);
+                }
+            }
+
+            void markMembersAsNotNull(string memberName, ref LocalState state)
+            {
+                foreach (Symbol member in CurrentSymbol.ContainingType.GetMembers(memberName))
```

Comment:
>GetMembers(memberName [](start = 71, length = 21)

This does not properly handle members declared in the base type and inherited into this type.

Reply:
This is the same lookup method that we use for `DefaultMember` attribute, as discussed in LDM.
I'll add tests to illustrate and capture the current (limited) behavior.

---
In reply to: [380879656](https://github.com/dotnet/roslyn/pull/41675#discussion_r380879656) [](ancestors = 380879656)